pkg/services/oracle/neofs/neofs.go

```diff
@@ -109,42 +112,55 @@ func getRange(ctx context.Context, c client.Client, addr *object.Address, ps ...
 	if err != nil {
 		return nil, err
 	}
-	data, err := c.ObjectPayloadRangeData(ctx, new(client.RangeDataParams).WithAddress(addr).WithRange(r))
+	res, err := c.ObjectPayloadRangeData(ctx, new(client.RangeDataParams).WithAddress(addr).WithRange(r))
 	if err != nil {
 		return nil, err
+	} else if err = neofsapistatus.ErrFromStatus(res.Status()); err != nil {
```

Comment:
Maybe we need some common wrapper to handle these cases?

Reply:
I can extract the error from the client's return in a separate function, then only one condition will remain. Or did you mean the type-wrapper over the client?

Reply:
As we don't care about this client/server error separation much it'd be a bit more convenient to have code (and there is a number of places doing this) checking some single error instead of two errors if possible.

Reply:
I've made a feature request for native support of "simple" clients in NeoFS SDK https://github.com/nspcc-dev/neofs-sdk-go/issues/91.

For now I'll implement the code for this purpose in private function.

Reply:
See https://github.com/nspcc-dev/neo-go/pull/2276/commits/a44f69be3bc31f9d1c64ee1cc1bebb63245498b6